### DISCUSSION TITLE
AAD support coming out of preview?

### DISCUSSION THREAD

**Discussion by jeingrah** [Azure/azure-cosmos-dotnet-v3#2320] [Q&A] [answered]
Hello,

I was wondering if there were current plans on bringing the support for AAD out of preview?

This PR last year added it https://github.com/Azure/azure-cosmos-dotnet-v3/pull/1798 and the constructor that allows passing in a TokenCredential is still in preview.  We currently have integration that uses endpoint/key but want to start moving to RBAC and without this we wouldn't be able to.

Thanks!

**Comment by j82w** [comment, marked as answer]
The current tentative plan is to GA AAD in June. The SDK will move it out of preview when the Cosmos DB service GAs the AAD integration.

**Reply by j82w** [reply]
@jeingrah AAD has officially GA. It's part of the 3.19.0 release.
